modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectValidationRuleLocalServiceImpl.java

```diff
@@ -480,6 +476,59 @@ private List<ObjectValidationRule> _getObjectValidationRules(
 		return objectValidationRules;
 	}
 
+	private List<ObjectValidationRuleSetting>
+			_updateObjectValidationRuleSettings(
+				ObjectValidationRule objectValidationRule,
+				List<ObjectValidationRuleSetting>
+					newObjectValidationRuleSettings)
+		throws PortalException {
+
+		Set<ObjectValidationRuleSetting> objectValidationRuleSettings =
+			SetUtil.fromCollection(newObjectValidationRuleSettings);
+
+		for (ObjectValidationRuleSetting oldObjectValidationRuleSetting :
+				_objectValidationRuleSettingPersistence.
+					findByObjectValidationRuleId(
+						objectValidationRule.getObjectValidationRuleId())) {
+
+			boolean delete = true;
+
+			for (ObjectValidationRuleSetting newObjectValidationRuleSetting :
+					newObjectValidationRuleSettings) {
+
+				if (StringUtil.equals(
+						oldObjectValidationRuleSetting.getName(),
+						newObjectValidationRuleSetting.getName()) &&
+					StringUtil.equals(
+						oldObjectValidationRuleSetting.getValue(),
+						newObjectValidationRuleSetting.getValue())) {
+
+					delete = false;
+					objectValidationRuleSettings.remove(
+						newObjectValidationRuleSetting);
```

Comment:
@devpedrotavares I think we could add a `break;` here so we don't need to check all elements